File: 241216/삼각형 컨베이어 벨트/conveyor-belt-triangle.js
let fs = require('fs');
let [temp1, ...temp2] = fs.readFileSync(0).toString().trim().split('\n');

const [n ,t] = temp1.split(' ').map(Number);
const arr = temp2.map(val => val.split(' ', n).map(Number));


for(let i = 0; i < t; i++){
    const box1 = arr[0].pop();
    const box2 = arr[1].pop();
    const box3 = arr[2].pop();

    arr[0].unshift(box3);
    arr[1].unshift(box1);
    arr[2].unshift(box2);
}
console.log(arr[0].join(' '));
console.log(arr[1].join(' '));
console.log(arr[2].join(' '));

File: 260205/삼각형 컨베이어 벨트/conveyor-belt-triangle.py
n, t = map(int, input().split())

l = input().split()
r = input().split()
d = input().split()

belt = l + r + d
length = 3 * n

t %= length
belt = belt[-t:] + belt[:-t]

l = belt[:n]
r = belt[n:2*n]
d = belt[2*n:]
print(' '.join(l))
print(' '.join(r))
print(' '.join(d))